source/extensions/filters/http/compressor/compressor_filter.cc

```diff
@@ -0,0 +1,24 @@
+#include "extensions/filters/http/compressor/compressor_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Compressor {
+
+CompressorFilterConfig::CompressorFilterConfig(
+    const envoy::extensions::filters::http::compressor::v3::Compressor& generic_compressor,
+    const std::string& stats_prefix, Stats::Scope& scope, Runtime::Loader& runtime,
+    CompressorFactoryPtr compressor_factory)
+    : Common::Compressors::CompressorFilterConfig(
+          generic_compressor, stats_prefix + compressor_factory->statsPrefix(), scope, runtime,
+          compressor_factory->contentEncoding()),
+      compressor_factory_(std::move(compressor_factory)) {}
+
+std::unique_ptr<Envoy::Compressor::Compressor> CompressorFilterConfig::makeCompressor() {
```

Comment:
qq: we were thinking about why we would lazily create the compressor (and thus have the config factory create a compressor factory) vs creating the compressor to begin with?

Reply:
Compressors are created (or not) for every new stream if multiple conditions are met.

This particular wrapper method is going to get more meat when the `gzip` filter is dropped and the functionality of `Common::Compressors::CompressorFilterConfig` is moved to this class.

Reply:
> Compressors are created (or not) for every new stream if multiple conditions are met.

Yep, this became clear as I read the rest of the PR. Sorry, I meant to delete this comment.